generators/client/templates/angular/src/main/webapp/app/layouts/main/_main.component.ts

```diff
@@ -50,6 +50,15 @@ export class <%=jhiPrefixCapitalized%>MainComponent implements OnInit {
                     params = destinationEvent.params;
                     destinationData = destinationEvent.data;
                     destinationName = destinationEvent.url[0].path;
+                } else {
+                    destinationEvent = event.state.root.children[0];
```

Comment:
I'll test another path, but I think it's for every protected paths.. I think it's because sometimes our routes have an abstract parent, as admin routes have the admin state that protect them, but others doesn't have any parent, like settings. That's why I did the if else thing above, but I agree, it's not clean.. :disappointed: